LoopTests/Mock Stores/MockGlucoseStore.swift

```diff
@@ -0,0 +1,181 @@
+    
+    var testType: DataManagerTestType
+    
+    var latestGlucose: GlucoseSampleValue? {
+        return StoredGlucoseSample(
+            sample: HKQuantitySample(
+                type: HKQuantityType.quantityType(forIdentifier: .bloodGlucose)!,
+                quantity: HKQuantity(unit: HKUnit.milligramsPerDeciliter, doubleValue: latestGlucoseValue),
+                start: glucoseStartDate,
+                end: glucoseStartDate
+            )
+        )
+    }
+    
+    var preferredUnit: HKUnit?
+    
+    var sampleType: HKSampleType = HKSampleType.quantityType(forIdentifier: HKQuantityTypeIdentifier.bloodGlucose)!
+    
+    var delegate: GlucoseStoreDelegate?
+    
+    var managedDataInterval: TimeInterval?
+    
+    var authorizationRequired: Bool = false
+    
+    var sharingDenied: Bool = false
+    
+    var healthStore: HKHealthStore = HKHealthStoreMock()
+    
+    func authorize(toShare: Bool, _ completion: @escaping (HealthKitSampleStoreResult<Bool>) -> Void) {
+        completion(.success(true))
+    }
+    
+    func addGlucose(_ glucose: NewGlucoseSample, completion: @escaping (GlucoseStoreResult<GlucoseValue>) -> Void) {
+        // Using the dose store error because we don't need to create GlucoseStore errors just for the mock store
+        completion(.failure(DoseStore.DoseStoreError.configurationError))
+    }
+    
+    func addGlucose(_ values: [NewGlucoseSample], completion: @escaping (GlucoseStoreResult<[GlucoseValue]>) -> Void) {
+        // Using the dose store error because we don't need to create GlucoseStore errors just for the mock store
+        completion(.failure(DoseStore.DoseStoreError.configurationError))
+    }
+    
+    func getCachedGlucoseSamples(start: Date, end: Date?, completion: @escaping ([StoredGlucoseSample]) -> Void) {
+        completion([latestGlucose as! StoredGlucoseSample])
+    }
+    
+    func generateDiagnosticReport(_ completion: @escaping (String) -> Void) {
+        completion("")
+    }
+    
+    func purgeGlucoseSamples(matchingCachePredicate cachePredicate: NSPredicate?, healthKitPredicate: NSPredicate, completion: @escaping (Bool, Int, Error?) -> Void) {
+        // Using the dose store error because we don't need to create GlucoseStore errors just for the mock store
+        completion(false, 0, DoseStore.DoseStoreError.configurationError)
+    }
+    
+    func executeGlucoseQuery(fromQueryAnchor queryAnchor: GlucoseStore.QueryAnchor?, limit: Int, completion: @escaping (GlucoseStore.GlucoseQueryResult) -> Void) {
+        // Using the dose store error because we don't need to create GlucoseStore errors just for the mock store
+        completion(.failure( DoseStore.DoseStoreError.configurationError))
+    }
+    
+    func counteractionEffects<Sample>(for samples: [Sample], to effects: [GlucoseEffect]) -> [GlucoseEffectVelocity] where Sample : GlucoseSampleValue {
+        return [] // TODO: check if we'll ever want to test this
+    }
+    
```

Comment:
I didn't have these functions to return any meaningful values because they aren't used in the portion of the algo that I'm testing (but they could be updated later if needed for other LDM tests)